clang/test/CodeGenSYCL/functionptr-addrspace.cpp

```diff
@@ -21,5 +21,16 @@ int main() {
     invoke_function(r, &a);
     invoke_function(f, &a);
   });
+
+  // Test function pointer as kernel argument. Function pointers should have program address space i.e. 0.
+
+  int (*fptr)();
+  int *ptr;
+
+  // define dso_local spir_kernel void @{{.*}}fake_kernel_2{{.*}}(i32 ()* align 4 %_arg_fptr, i32 addrspace(1)* align 4 %_arg_ptr)
```

Comment:
Just an FYI, since April 2020 (or whenever this commit was pulled down to DPCPP project https://reviews.llvm.org/D77119), the address space has been 0, which is why there was a regression downstream after  https://reviews.llvm.org/D119045. If the correct behavior is some other address space, I think it should be done as a follow-up patch (if its not a quick fix). This patch is just fixing the regression

Reply:
https://github.com/intel/llvm/runs/5747268781?check_suite_focus=true <-  this failure is not related and fixed by https://github.com/intel/llvm/commit/fa054cc41f0e0359b3cc6db5dd63ef8e98c50fa4.

Reply:
OK thanks for letting me know. There is one more failure on Windows - SYCL :: Printf/mixed-address-space.cpp

I am trying to build a Windows workspace to check what is wrong.

Reply:
I don't think the failing test is due to this patch. I was unable to reproduce issue on my local system and so I took a closer look at the test - https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Printf/mixed-address-space.cpp. The kernel here isn't capturing anything. So I don't know why this patch would affect it (since the change in this patch only applies to function pointer kernel arguments)

Reply:
> since April 2020 (or whenever this commit was pulled down to DPCPP project https://reviews.llvm.org/D77119), the address space has been 0, which is why there was a regression downstream after https://reviews.llvm.org/D119045.

Thanks, this makes me a bit more confident about what the patch does.

Reply:
> Being honest I'm not sure what is the correct address space for function pointers in kernel arguments. It would be great to get feedback from @AlexeySachkov .

At SYCL level, we don't plan to have function pointers as kernel arguments